### PR TITLE
monitoring: align with ceph/ceph prometheus alerts

### PR DESCRIPTION
**Pull request by ahgraber**
Update rook prometheus localrules files to align with ceph/ceph prometheus alerts

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by ahgraber** [comment]
@travisn 

This PR replaces my git travesty of #11585

**Comment by travisn** [comment]
Looks like the entire rules file changed, at least with whitespace. Is this expected or can we avoid all the whitespace update to more easily see the diff? Maybe it's expected, I didn't look very close, but want to understand.

**Comment by ahgraber** [comment]
> Looks like the entire rules file changed, at least with whitespace. Is this expected or can we avoid all the whitespace update to more easily see the diff? Maybe it's expected, I didn't look very close, but want to understand.

It's both whitespace and enforced quotes around strings.  I'm assuming this is due to a linting process and/or whatever function is used to migrate the jsonnet to yaml.

I would be happy to try to go back and clean it up, but if the ceph/ceph process stays the same, then manual cleanup may be required every time a copy/paste is made to pull the changes from ceph into rook.

**Comment by ahgraber** [comment]
I've gone back to sort and clean up old/new versions manually so everything is relatively easy to compare. Unfortunately, because this necessitated sorting both the old and new files, it's not really commit-ready.

Here is a high-level summary of the changes:

- No alerts were removed.
- New alerts were added:
   - nodes.CephNodeNetworkBondDegraded
   - healthchecks.CephDaemonSlowOps
   - cephadm.*
   - PrometheusServer.*
- Minor updates/clarifications to alert descriptions.
- Updates to rule expressions:
   - nodes.CephNodeNetworkPacketDrops
   - nodes.CephNodeNetworkPacketErrors
   - nodes.CephNodeInconsistentMTU
   - pools.CephPoolGrowthWarning

**Comment by travisn** [comment]
Thanks for the summary! No need to remove the whitespace changes since they were in the ceph repo. The less we diverge the better.

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml [hunk 1]
```diff
@@ -1,838 +1,655 @@
...
+        for: "30s"
+        expr: "ceph_daemon_health_metrics{type=\"SLOW_OPS\"} > 0"
+        labels:
+          severity: 'warning'
+          type: 'ceph_default'
+        annotations:
+          summary: "{{ $labels.ceph_daemon }} operations are slow to complete"
+          description: "{{ $labels.ceph_daemon }} operations are taking too long to process (complaint time exceeded)"
+          documentation: "https://docs.ceph.com/en/latest/rados/operations/health-checks#slow-ops"
+  - name: "cephadm"
```
Comment (travisn) [inline deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml:586]: Let's remove all of the cephadm alerts since they don't apply to Rook.

Comment (ahgraber) [reply deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml]: `cephadm` ruleset removed.

Should I squash to a single commit?

**Review by travisn — CHANGES_REQUESTED** [review]
> Should I squash to a single commit?

Keeping a separate commit looks good. Can you just change the commit description to explain the removal of cephadm rules, instead of using the same commit message as the original one? Then I think we're good to merge, thanks!

**Comment by ahgraber** [comment]
> change the commit description to explain the removal of cephadm rules

I took this to mean change the description body but not the title.  LMK if I also need to update the title.

**Comment by travisn** [comment]
> > change the commit description to explain the removal of cephadm rules
> 
> I took this to mean change the description body but not the title. LMK if I also need to update the title.

Sorry, please also update the title to differentiate the commits.

**Review by travisn — APPROVED** [review]

**Pull request merged**